go/vt/wrangler/testlib/migrate_served_types_test.go

```diff
@@ -236,3 +248,355 @@ func TestMigrateServedTypes(t *testing.T) {
+	vp := NewVtctlPipe(t, ts)
+	defer vp.Close()
+
+	// create keyspace
+	if err := ts.CreateKeyspace(context.Background(), "ks", &topodatapb.Keyspace{
+		ShardingColumnName: "keyspace_id",
+		ShardingColumnType: topodatapb.KeyspaceIdType_UINT64,
+	}); err != nil {
+		t.Fatalf("CreateKeyspace failed: %v", err)
+	}
+
+	// create the first source shard
+	source1Master := NewFakeTablet(t, wr, "cell1", 20, topodatapb.TabletType_MASTER, nil,
+		TabletKeyspaceShard(t, "ks", "-80"))
+	source1Replica := NewFakeTablet(t, wr, "cell1", 21, topodatapb.TabletType_REPLICA, nil,
+		TabletKeyspaceShard(t, "ks", "-80"))
+	source1Rdonly := NewFakeTablet(t, wr, "cell1", 22, topodatapb.TabletType_RDONLY, nil,
+		TabletKeyspaceShard(t, "ks", "-80"))
+
+	// create the second source shard
+	source2Master := NewFakeTablet(t, wr, "cell1", 30, topodatapb.TabletType_MASTER, nil,
+		TabletKeyspaceShard(t, "ks", "80-"))
+	source2Replica := NewFakeTablet(t, wr, "cell1", 31, topodatapb.TabletType_REPLICA, nil,
+		TabletKeyspaceShard(t, "ks", "80-"))
+	source2Rdonly := NewFakeTablet(t, wr, "cell1", 32, topodatapb.TabletType_RDONLY, nil,
+		TabletKeyspaceShard(t, "ks", "80-"))
+
+	dest1Master := NewFakeTablet(t, wr, "cell1", 40, topodatapb.TabletType_MASTER, nil,
+		TabletKeyspaceShard(t, "ks", "-40"))
+	dest1Replica := NewFakeTablet(t, wr, "cell1", 41, topodatapb.TabletType_REPLICA, nil,
+		TabletKeyspaceShard(t, "ks", "-40"))
+	dest1Rdonly := NewFakeTablet(t, wr, "cell1", 42, topodatapb.TabletType_RDONLY, nil,
+		TabletKeyspaceShard(t, "ks", "-40"))
+
+	//	create the second source shard
+	dest2Master := NewFakeTablet(t, wr, "cell1", 50, topodatapb.TabletType_MASTER, nil,
+		TabletKeyspaceShard(t, "ks", "40-80"))
+	dest2Replica := NewFakeTablet(t, wr, "cell1", 51, topodatapb.TabletType_REPLICA, nil,
+		TabletKeyspaceShard(t, "ks", "40-80"))
+	dest2Rdonly := NewFakeTablet(t, wr, "cell1", 52, topodatapb.TabletType_RDONLY, nil,
+		TabletKeyspaceShard(t, "ks", "40-80"))
+
+	dest3Master := NewFakeTablet(t, wr, "cell1", 60, topodatapb.TabletType_MASTER, nil,
+		TabletKeyspaceShard(t, "ks", "80-c0"))
+	dest3Replica := NewFakeTablet(t, wr, "cell1", 61, topodatapb.TabletType_REPLICA, nil,
+		TabletKeyspaceShard(t, "ks", "80-c0"))
+	dest3Rdonly := NewFakeTablet(t, wr, "cell1", 62, topodatapb.TabletType_RDONLY, nil,
+		TabletKeyspaceShard(t, "ks", "80-c0"))
+
+	// create the second source shard
+	dest4Master := NewFakeTablet(t, wr, "cell1", 70, topodatapb.TabletType_MASTER, nil,
+		TabletKeyspaceShard(t, "ks", "c0-"))
+	dest4Replica := NewFakeTablet(t, wr, "cell1", 71, topodatapb.TabletType_REPLICA, nil,
+		TabletKeyspaceShard(t, "ks", "c0-"))
+	dest4Rdonly := NewFakeTablet(t, wr, "cell1", 72, topodatapb.TabletType_RDONLY, nil,
+		TabletKeyspaceShard(t, "ks", "c0-"))
+
+	if source1Master == nil || source1Replica == nil || source1Rdonly == nil || source2Master == nil || source2Replica == nil || source2Rdonly == nil {
```

Comment:
@rafael what's the idea of these if statements? staticcheck is complaining.

Reply:
Ooops - I missed this.

I added this for debugging purposes. I was adding some panic here when the code was not fully ready. Then forgot to remove this conditions. I created this PR to address this: 

https://github.com/vitessio/vitess/pull/4753